### PR TITLE
chore(clients): add workflow to auto-close PR

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-2970

### Changes included:

heavily inspired by https://github.com/algolia/doc/blob/master/.github/workflows/dont-edit-apis.yml, we can kindly ask the user to redirect their contributions to the automation repository, or ask them to open an issue if they don't want to.

### CODE REVIEW

File: templates/do-not-edit-this-repository.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    name: Close PR
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Close PR
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+        run: |
```
Comment (millotp) [inline templates/do-not-edit-this-repository.yml:23]: you need to check that the PR is targeting, because it's possible to do bug fixes on older version

**Review by millotp — CHANGES_REQUESTED** [review]
nice

File: templates/do-not-edit-this-repository.yml (hunk 1 above)
Comment (shortcuts) [reply templates/do-not-edit-this-repository.yml:23]: yup this should do it https://github.com/algolia/api-clients-automation/blob/b227ed49c43e9fe2ad4b05a1a64acff5a12b07d7/templates/do-not-edit-this-repository.yml#L3

File: generators/src/main/java/com/algolia/codegen/AlgoliaDartGenerator.java [hunk 2]
```diff
@@ -104,6 +104,9 @@ public void processOpts() {
     supportingFiles.add(new SupportingFile("LICENSE", "../client_core/", "LICENSE"));
     supportingFiles.add(new SupportingFile("issue.yml", "../../.github/workflows", "issue.yml"));
     supportingFiles.add(new SupportingFile("Bug_report.yml", "../../.github/ISSUE_TEMPLATE", "Bug_report.yml"));
+    supportingFiles.add(
```
Comment (millotp) [inline generators/src/main/java/com/algolia/codegen/AlgoliaDartGenerator.java:107]: we could have an helper to add the 4 common files, with the root folder as argument

File: templates/do-not-edit-this-repository.yml (hunk 1 above)
Comment (millotp) [reply templates/do-not-edit-this-repository.yml:23]: oh sorry I missed that

**Comment by shortcuts** [comment]
> Greatest achievement of my life. Someone was inspired by something I did 🤣.

<3

File: generators/src/main/java/com/algolia/codegen/AlgoliaDartGenerator.java (hunk 2 above)
Comment (shortcuts) [reply generators/src/main/java/com/algolia/codegen/AlgoliaDartGenerator.java]: https://github.com/algolia/api-clients-automation/pull/3834/commits/a589dfa8f1088a1f84691a449273f2a9bfff6780 hopla

**Review by millotp — APPROVED** [review]
beautiful

**Pull request merged**